src/errors.ts

```diff
@@ -0,0 +1,42 @@
+export class PetraApiError extends Error {
+  constructor(
+    readonly code: number,
+    readonly status: string,
+    message: string,
+  ) {
+    super(message);
+    this.status = status;
+    this.code = code;
+    Object.setPrototypeOf(this, PetraApiError.prototype);
+  }
+}
+
+export const PetraApiErrors = Object.freeze({
```

Comment:
wouldnt it be easier for Petra to return those error objects? instead of only the code and have the plugin hold a map here?